include/mc_rbdyn/Collision.h

```diff
@@ -24,18 +24,23 @@ struct MC_RBDYN_DLLAPI Collision
             double i,
             double s,
             double d,
-            const std::vector<std::string> & r1Joints = {},
-            const std::vector<std::string> & r2Joints = {})
-  : body1(b1), body2(b2), iDist(i), sDist(s), damping(d), r1Joints(r1Joints), r2Joints(r2Joints)
+            const std::vector<std::string> & r1ActiveJoints = {},
+            const std::vector<std::string> & r2ActiveJoints = {},
+            const std::vector<std::string> & r1UnactiveJoints = {},
+            const std::vector<std::string> & r2UnactiveJoints = {})
```

Comment:
```suggestion
            std::optional<std::vector<std::string>> r1Joints = {},
            std::optional<std::vector<std::string>> r2Joints = {},
            bool r1JointsInactive = false,
            bool r2JointsInactive = false)
```

Here `r1JointsInactive` is used to disambiguate the meaning of `r1Joints` if it is provided

I initially introduced these active joints before mc_rtc 2.0.0 and thus before C++17 adoption. Thinking back on the design, it's better to treat an explicitly empty joints entry as meaning "no active joints" rather than "all joints active". This is easier to express with `std::optional`

This way:
- r1Joints can be treated as r1ActiveJoints in the yaml
- r1ActiveJoints is the same path as before
- r1InactiveJoints is r1Joints non-empty + true for `r1JointsInactive`
- we can treat an empty vector for r1Joints (+ `r1JointsInactive == true`)

(same with r2)